added tests to appInfo route and some mocking tests

rubenmarques-dev opened the pull request:
Implemented tests to check the return of /api/appInfo route.

Added a "test" module called user to explore some assertings and the mocking feature of jest.

Changed in package.json:

"scripts": {
-    "start": "PORT=3001 node ./bin/www",
-    "test": "NODE_ENV=test jest"
+    "start": "node ./bin/www",
+    "test": "jest"
   },

Commit:
added tests to appInfo route and some mocking tests

Commit:
Merge branch 'develop' into backend-tests

Commit:
removed some comments

Commit:
Merge branch 'backend-tests' of https://github.com/cubicon-feup/mygrant into backend-tests

Commit:
configure frontend tests and add dummy component to s

edgarlpassos commented:
Há conflitos no package.json
Porque é que foi retirado o PORT=3001? É necessário para ter o React e o Express a correr ao mesmo tempo (em portas diferentes).

diogohmcruz commented:
@edgarlpassos nao precisas de ter o port=3001, meti o default do express a ser a 3001

diogohmcruz committed:
Merge branch 'develop' into backend-tests

diogohmcruz committed:
Separated tests

Agora ao correr os testes do server não corre os do client.
Os ficheiros babel também não me parecem precisos.
(Embora o branch se chame backend-tests, também tem tests de frontend)

diogohmcruz committed:
Merge branch 'develop' into backend-tests
